sdk/core/System.ClientModel/src/Pipeline/ClientPipeline.cs

```diff
@@ -157,7 +188,14 @@ public static ClientPipeline Create(
         int beforeTransportIndex = index;
 
         // Add the transport.
-        policies[index++] = options.Transport ?? HttpClientPipelineTransport.Shared;
+        if (loggingOptions == null || loggingOptions.ShouldUseDefaultPipelineTransport())
+        {
+            policies[index++] = options.Transport ?? HttpClientPipelineTransport.Shared;
+        }
+        else
+        {
+            policies[index++] = options.Transport ?? new HttpClientPipelineTransport(null, loggingOptions.EnableLogging ?? ClientLoggingOptions.DefaultEnableLogging, loggingOptions.LoggerFactory);
```

Comment:
This is a downside of logging from the transport - can't share if using ILogger

Reply:
I think that's OK -- the same rules would apply to the transport as any other policy: https://gist.github.com/annelo-msft/eea9b66a3545ee19cbe81dfb3a2e9b59#clientpipelinecreate.  The biggest consideration here is that we have a single instance of HttpClient that is shared across client instances -- I expect we can achieve that even if there are different ILoggers per client.